test/framework/kogitoapp.go

```diff
@@ -209,8 +212,56 @@ func setupBuildImageStreams(kogitoApp *v1alpha1.KogitoApp) {
 	// If "KOGITO_BUILD_IMAGE_STREAM_TAG" is defined, it is taken into account
 	// If not defined then search for specific s2i and runtime tags
 	// If none, let the operator manage
-	kogitoApp.Spec.Build.ImageS2ITag = config.GetBuildS2IImageStreamTag()
-	kogitoApp.Spec.Build.ImageRuntimeTag = config.GetBuildRuntimeImageStreamTag()
+	kogitoApp.Spec.Build.ImageS2ITag = getBuildS2IImageStreamTag(kogitoApp)
+	kogitoApp.Spec.Build.ImageRuntimeTag = getBuildRuntimeImageStreamTag(kogitoApp)
+
 	// If "KOGITO_BUILD_IMAGE_VERSION" is defined, it's taken into account, otherwise set the current version
 	kogitoApp.Spec.Build.ImageVersion = config.GetBuildImageVersion()
 }
+
+func getBuildRuntimeImageStreamTag(kogitoApp *v1alpha1.KogitoApp) string {
+	if len(config.GetBuildRuntimeImageStreamTag()) > 0 {
+		return config.GetBuildRuntimeImageStreamTag()
+	}
+
+	if isBuildImageRegistryOrNamespaceSet() {
+		buildType := resource.BuildTypeRuntime
+		if kogitoApp.Spec.Runtime == v1alpha1.QuarkusRuntimeType && !kogitoApp.Spec.Build.Native {
+			buildType = resource.BuildTypeRuntimeJvm
+		}
+
+		return getBuildImage(resource.BuildImageStreams[buildType][kogitoApp.Spec.Runtime])
+	}
+
+	return ""
+}
+
+func getBuildS2IImageStreamTag(kogitoApp *v1alpha1.KogitoApp) string {
+	if len(config.GetBuildS2IImageStreamTag()) > 0 {
+		return config.GetBuildS2IImageStreamTag()
+	}
+
+	if isBuildImageRegistryOrNamespaceSet() {
+		return getBuildImage(resource.BuildImageStreams[resource.BuildTypeS2I][kogitoApp.Spec.Runtime])
+	}
+
+	return ""
+}
+
+func isBuildImageRegistryOrNamespaceSet() bool {
+	return len(config.GetBuildImageRegistry()) > 0 || len(config.GetBuildImageNamespace()) > 0
+}
+
+func getBuildImage(image string) string {
+	registry := config.GetBuildImageRegistry()
```

Comment:
why not using the `Image` struc from operator framework package and use the convert method at the end ?

Reply:
Do you mean, to use this method framework.ConvertImageToImageTag(image), right? Will change it.

Reply:
yes that is right.
That would avoid any mistake when building the image name :)

Reply:
Changed